### PR TITLE
Accept file `URL` instances

### LINKED ISSUE #36: Accept file `URL` instances
Currently, the `makeDir` function doesn't accept a file `URL` instance:

```js
import makeDir from 'make-dir';

await makeDir(new URL('foo/', import.meta.url));
```

Results in the following error:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received an instance of URL
```

This makes `make-dir` harder to work with within Node.js ESM modules.

Users will run into this gotcha more frequently as most Node.js APIs that work with filesystem paths accept file `URL` instances, including [`mkdir`](https://nodejs.org/api/fs.html#fs_fspromises_mkdir_path_options).

In the meantime, we have to do this workaround:

```js
import { fileURLToPath } from 'url';
import makeDir from 'make-dir';

await makeDir(fileURLToPath(new URL('foo/', import.meta.url)));
```

### PR DESCRIPTION
**Pull request by jopemachine**
Fixes #36.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -43,6 +44,10 @@ const permissionError = pth => {
 	return error;
 };
 
+const getTargetDir = input => {
+	return typeof input === 'string' ? path.resolve(input) : input;
```
Comment (sindresorhus) [inline index.js:48]: What if the URL is relative? This is not the correct way to handle `URL`.

File: index.js [hunk 2]
```diff
@@ -51,7 +56,7 @@ const makeDir = async (input, options) => {
 	const stat = promisify(options.fs.stat);
 
 	if (useNativeRecursiveOption && options.fs.mkdir === fs.mkdir) {
-		const pth = path.resolve(input);
+		const pth = getTargetDir(input);
```
Comment (sindresorhus) [inline index.js:59]: The `.resolve` call is moot anyway here I think as the native method resolves internally.

**Comment by sindresorhus** [comment]
You're not handling `URL` where it's actually needed, on line 106.

**Comment by jopemachine** [comment]
Removed moot function call and added some tests for relative url.
It seems working for relative urls.
I would appreciate your letting me know if I got wrong :)

**Comment by sindresorhus** [comment]
> You're not handling URL where it's actually needed, on line 106.

^

**Comment by jopemachine** [comment]
Done. I'm sorry for the missing.

File: index.js [hunk 3]
```diff
@@ -32,6 +36,10 @@ const processOptions = options => {
 	};
 };
 
+const getTargetDir = input => {
+	return path.resolve(typeof input === 'string' ? input : input.pathname);
```
Comment (sindresorhus) [inline index.js:40]: This is incorrect. The Node.js docs even point that out: https://nodejs.org/api/url.html#urlfileurltopathurl

File: index.js [hunk 4]
```diff
@@ -98,7 +104,7 @@ const makeDir = async (input, options) => {
 		}
 	};
 
-	return make(path.resolve(input));
+	return make(getTargetDir(input));
```
Comment (sindresorhus) [inline index.js:107]: Don't use abbreviations.

File: index.js (hunk 3 above)
Comment (jopemachine) [reply index.js:40]: > This is incorrect. The Node.js docs even point that out: [nodejs.org/api/url.html#urlfileurltopathurl](https://nodejs.org/api/url.html#urlfileurltopathurl)

Thanks for letting me know. It seems I need more study Node modules.

Could I ask if we could use `fileURLToPath` here?
The document says it was added in 10.12.0, but this module's target node engines is greater than or equal to 10.
Is it acceptable to change this module's engine to 10.12.0?
Or do I need to add some polyfills to replace `fileURLToPath`?

Comment (sindresorhus) [reply index.js:40]: It's ok to use `fileURLToPath`. I plan to target Node.js 14 soon.

File: index.js [hunk 5]
```diff
@@ -32,6 +37,10 @@ const processOptions = options => {
 	};
 };
```
Comment (sindresorhus) [inline index.js:39]: Use a helper method for readability:

```js
const toPath = urlOrPath => urlOrPath instanceof URL ? fileURLToPath(urlOrPath) : urlOrPath;
```

**Comment by sindresorhus** [comment]
Tests were failing on Windows.

**Pull request closed without merging**